tensorflow/core/common_runtime/mkl_layer_registry.h

```diff
@@ -0,0 +1,107 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_CORE_COMMON_RUNTIME_MKL_LAYER_REGISTRY_H_
+#define TENSORFLOW_CORE_COMMON_RUNTIME_MKL_LAYER_REGISTRY_H_
+
+#ifdef INTEL_MKL
+
+#include <set>
+#include <map>
+#include <utility>
+#include <string>
+
+#include "tensorflow/core/framework/types.pb.h"
+
+namespace tensorflow {
+
+// A global MKLLayerRegistry is used to hold information about all MKL-compliant
+// layers. Registry maintains information about <opname,T> pair supported
+// by MKL API.
+class MklLayerRegistry {
```

Comment:
Right, but could just just use the OpKernel registry that we already have? It can be queried in this way:

https://github.com/tensorflow/tensorflow/blob/99fe61a8a8f3dd41b4e1e4dedfc53b45f67e88a7/tensorflow/core/framework/op_kernel.h#L1062